restorers/model/nafnet/nafblock.py

```diff
@@ -0,0 +1,90 @@
+import tensorflow as tf
+from tensorflow import keras
+
+
+class SimpleGate(keras.layers.Layer):
+
+    def __init__(self, factor=2):
+        super().__init__()
+        self.factor = factor
+
+    def call(self, x):
+        x_list = tf.split(x, num_or_size_splits=self.factor, axis=-1)
+        result = x_list[0]
+        for i in range(1, len(x_list)):
+            result *= x_list[i]
+        return result
+
+
+class SimplifiedChannelAttention(keras.layers.Layer):
+    def __init__(self, channels, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.channels = channels
+        self.avg_pool = keras.layers.GlobalAveragePooling2D()
+        self.conv = keras.layers.Conv2D(filters=channels, kernel_size=1)
+
+    def call(self, inputs):
+        average_pooling = self.avg_pool(inputs)
+        feature_descriptor = tf.reshape(average_pooling, shape=(-1, 1, 1, self.channels))
+        features = self.conv(feature_descriptor)
+        return inputs * features
+
+
+class NAFBlock(keras.layers.Layer):
```

Comment:
Please add docstrings like [this](https://github.com/soumik12345/restorers/blob/main/restorers/model/zero_dce/dce_layer.py#L9).